### ISSUE TITLE
Improve TimingSystem API

### ISSUE THREAD

**Issue by Pigalala** [Makkuusen/TimingSystem#337]
would be nice to extend the API to improve support for external plugins. 

We could do some research into other plugins and see how their APIs are implemented.

**Comment by Makkuusen** [comment]
Yeah, I've been thinking the same before. I guess the difficult part is to know what would be useful for people.

I feel like it might have to be developed after demands if we wanna be efficient about it. Normally I try to add things I find useful when working on my own additions, unless it is way to complicated.

But like if we could make a list of the things that we think would be useful, we could start planning a v2 of the API.
